src/ansi-c/c_typecheck_base.cpp

```diff
@@ -743,11 +754,14 @@ void c_typecheck_baset::typecheck_declaration(
       typecheck_spec_expr(static_cast<codet &>(contract), ID_C_spec_ensures);
       parameter_map.clear();
 
+      if(contract.find(ID_C_spec_assigns).is_not_nil())
+        new_symbol.type.add(ID_C_spec_assigns) =
+          contract.find(ID_C_spec_assigns);
```

Comment:
Is repeating the `find` expensive?

Reply:
Good catch; it may be. Duplicates have been removed.